mautic/mautic
# Log segment and campaign change history to give visibility

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This will log when a contact is added or removed from campaigns and segments to give visibility into what is going on. 

[//]: # ( As applicable: )
#### Steps to test this PR:
1. Add and remove contacts to campaigns and segments then remove them. 
2. Check the contact timelines and there should be an entry for the event
3. Test manually adding and removing a contact directly but also test building a segment that is associated with a campaign.
